src/libraries/Common/src/Interop/Unix/libc/Interop.GetParentPid.cs

```diff
@@ -0,0 +1,13 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class libc
+    {
+        [GeneratedDllImport(Libraries.Libc, EntryPoint = "getppid")]
+        internal static partial int GetParentPid();
```

Comment:
It would be nice to add this file next to https://github.com/dotnet/runtime/blob/e46ae80b00eab500bd93a86c8077a0595ce79de0/src/libraries/Common/src/Interop/Unix/System.Native/Interop.GetPid.cs#L10-L11 and export `SystemNative_GetParentPid` symbol from System.Native (see: https://grep.app/search?q=SystemNative_GetPid&regexp=true&filter[repo][0]=dotnet/runtime). If I remember it correctly, we intentionally try not to import symbol from libc directly (except for tests). `getppid` was indeed an exceptional case.

Reply:
We try not to use `SystemNative_*` APIs from out-of-box assemblies like the `Microsoft.Extensions.*` assemblies as they're considered private APIs for the Microsoft.NETCore.App framework.

Reply:
I saw that `libc` was once removed from Interop.Libraries.cs https://github.com/dotnet/runtime/commit/27bc21d46946f7fde41163ef67f1253c4bffe71d#diff-c1db2d15cd6bf99da7a388eca0ae209ff70f2708d8c4ee6f10d11ad320bdb917, not sure if it makes sense to add it back for OOTB assemblies. (e.g. System.Drawing interop code is still maintained under its own hierarchy)

Reply:
> We try not to use SystemNative_* APIs from out-of-box assemblies like the Microsoft.Extensions.* assemblies as they're considered private APIs for the Microsoft.NETCore.App framework.

It is even more of a stronger statement than "try not to". We explicitly **can't**. This is because the `Microsoft.Extensions.*` assemblies target netfx where the System.Native shim isn't present. (Note the netfx assembly would also be used on traditional Mono). And also new Extension assemblies can run on older .NET Core versions. In that case the System.Native shim will be present, but it won't have the new shim method.

So for assemblies that aren't in the Microsoft.NETCore.App shared framework, we can't use the System.Native shims.